File: InC/p6.c
/*
 * Project Euler number 6
 *
 * Find the sum of squares square of sums diff for first 100 natural numbers
 */

#include <stdio.h>

int sumSquares() {

	int i;
	long int sumSquares = 0;

	for(i = 1; i <= 100; i++){
		sumSquares += (i * i);
	}

	return sumSquares;
}

int squareSums() {

	int i;
	int nums = 0;
	long squareSums = 0;

	for(i = 1; i <= 100; i++) {
		nums += i;
	}

	return squareSums;
}

int main() {

	long answer = squareSums() - sumSquares();
	printf("Project Euler 6 solution: %ld\n", answer);

	return 0;
}

File: InPython/p6.py
"""
Project Euler Problem 6
"""

def sum_squares():
    i = 0
    sumSquares = 0

    while i <= 100:
        sumSquares += (i * i)
        i += 1

    return sumSquares

def square_sums():
    i = 0
    nums = 0
    squareSums = 0

    while i <= 100:
        nums += i
        i += 1
    squareSums = nums * nums
    return squareSums

def main():
    print "Project Euler 6 solution: " +  str(square_sums() - sum_squares())

if __name__ == '__main__':
    main()
